### ISSUE TITLE
[BUG][JAVA][FEIGN] Doesn't compile when generateSupportingFiles=false

### ISSUE THREAD

**Issue by juherr** [OpenAPITools/openapi-generator#9925]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The generated client interfaces import `ApiClient` and `EncodingUtils` which are not generated when `generateSupportingFiles=false`

##### openapi-generator version

5.2.0

##### Suggest a fix

* Generate `EncodingUtils` event if `generateSupportingFiles=false`
* Don't `extends ApiClient.Api` if `generateSupportingFiles=false`

**Comment by juherr** [comment]
Ping @davidkiss 🙏

**Comment by davidkiss** [comment]
@juherr I worked on the feign integration many years ago and not familiar with the latest code.

**Comment by juherr** [comment]
@davidkiss No problem. Just notice you are the only one listed on https://github.com/OpenAPITools/openapi-generator#template-creator ;)

**Comment by dfjmax** [comment]
@juherr did you find any workaround for this?

**Comment by juherr** [comment]
@dfjmax No. A fix should be propose but I'm not using it anymore and it seems nobody officially supports it
